lib/Conversion/ArithExtToArith/ArithExtToArith.cpp

```diff
@@ -13,6 +14,20 @@ namespace arith_ext {
 #define GEN_PASS_DEF_ARITHEXTTOARITH
 #include "lib/Conversion/ArithExtToArith/ArithExtToArith.h.inc"
 
+template <typename ValueOrOpResult>
+TypedAttr modulusHelper(IntegerAttr mod, ValueOrOpResult op, bool mul = false) {
+  auto width = getElementTypeOrSelf(op).getIntOrFloatBitWidth();
+  auto modWidth = (mod.getValue() - 1).getActiveBits();
+  width = std::max(width, mul ? 2 * modWidth : modWidth + 1);
+  auto intType = IntegerType::get(op.getContext(), width);
+  auto truncmod = mod.getValue().sextOrTrunc(width);
+  if (auto st = mlir::dyn_cast_or_null<ShapedType>(op.getType())) {
```

Comment:
```suggestion
  if (auto st = mlir::dyn_cast<ShapedType>(op.getType())) {
```

I think here we won't ever expect `op.getType()` to be null!